Malmo/src/MissionSpec.cpp

```diff
@@ -18,656 +18,625 @@
     {
-        initialiser::initXSD();
-
-        xml_schema::properties props;
-        props.schema_location(xml_namespace, FindSchemaFile("Mission.xsd"));
-        
-        xml_schema::flags flags = xml_schema::flags::dont_initialize;
-        if( !validate )
-            flags = flags | xml_schema::flags::dont_validate;
+        std::istringstream is(xml);
+        boost::property_tree::read_xml(is, mission);
 
-        istringstream iss(xml);
-        this->mission = Mission_(iss, flags, props);
+        if (validate) {
+            // Agent side schema validation is lacking but Minecraft client will perform full validation (which is not optional).
+            const auto& xmlns = mission.get_optional<string>("Mission.<xmlattr>.xmlns");
+            if (xmlns == boost::none || MALMO_NAMESPACE != xmlns.get()) {
+                throw runtime_error("MissionSpec is invalid (namespace)");
+            }
+        }
     }
 
     std::string MissionSpec::getAsXML( bool prettyPrint ) const
     {
-        ostringstream oss;
-        
-        xml_schema::namespace_infomap map;
-        map[""].name = xml_namespace;
-        map[""].schema = "Mission.xsd";
+        std::ostringstream oss;
 
-        xml_schema::flags flags = xml_schema::flags::dont_initialize;
-        if( !prettyPrint )
-            flags = flags | xml_schema::flags::dont_pretty_print;
+        write_xml(oss, mission);
 
-        Mission_( oss, *this->mission, map, "UTF-8", flags );
-        
-        return oss.str();
+        std::string xml = oss.str();
+        if (!prettyPrint) 
+            xml.erase(std::remove(xml.begin(), xml.end(), '\n'), xml.end());
+        return xml;
     }
     
     // -------------------- settings for the server ---------------------------------
     
     void MissionSpec::setSummary( const std::string& summary )
     {
-        this->mission->About().Summary( summary );
+        mission.put("Mission.About.Summary", summary);
     }
 
     void MissionSpec::timeLimitInSeconds(float s)
     {
-        ServerHandlers::ServerQuitFromTimeUp_sequence& qq = this->mission->ServerSection().ServerHandlers().ServerQuitFromTimeUp();
-        if( !qq.size() ) 
-        {
-            // make a new ServerQuitFromTimeUp handler for the server_handlers
-            ServerQuitFromTimeUp q( s * MillisecondsInOneSecond );
-            qq.push_back( q );
-        }
-        else
-        {
-            ServerQuitFromTimeUp& q = qq.front();
-            q.timeLimitMs( s * MillisecondsInOneSecond );
-        }
+        mission.put("Mission.ServerSection.ServerHandlers.ServerQuitFromTimeUp.<xmlattr>.timeLimitMs", s);
     }
     
     void MissionSpec::createDefaultTerrain()
     {
-        this->mission->ServerSection().ServerHandlers().FlatWorldGenerator().reset();
-        DefaultWorldGenerator dwg;
-        this->mission->ServerSection().ServerHandlers().DefaultWorldGenerator( dwg );
+        forceWorldReset();
```

Comment:
See comment on forceWorldReset() below. (reset() and forceWorldReset() are different things!)